### PR TITLE
St archive fix

### LINKED ISSUE #518: TOTALPES value computed incorrectly on hyperthread capable platforms
Many HPC systems are hyperthread capable and can overload the number of tasks x threads per
node by factors of 2x or 4x or more.   The current setting of TOTALPES in cime fails to take this into account which results in an incorrect Model Cost estimate.   For example on yellowstone for a particular F case I use 960 tasks and 2 threads per task.  I have set the ptile to use 15 tasks per node so the number of tasks to be used for the cost estimate should be 1024.   Currently the 
TOTALPES value in cime for this run is 1920 and COST_PES=0.    They should be 960 and 1024.

### LINKED ISSUE #523: create_test --test-id [arg] needs to error if a component name is included in the arg string
If the create_test --test-id argument contains the name of a model component that is also listed in the config_archive.xml, then the st_archive will incorrectly move/copy files from the rundir to the DOUT_S_ROOT location because of the regex pattern matching. 

For example,
execca ./create_test --test-id b01_mosart_nomods ERR.f19_g16.B1850.yellowstone_intel.allactive-defaultio

Creates $CASE = b01_mosart_nomods ERR.f19_g16.B1850.yellowstone_intel.allactive-defaultio.b01_mosart_nomods

When the st_archive runs, it matches on the XML comp_archive_spec compname="mosart" for all
case files and incorrectly copies them all to $DOUT_S_ROOT/archive/$CASE/rof/hist. 

This issue maybe addressed by #500 which will include better pattern matching based on component output filename settings.

### LINKED ISSUE #532: ./create_test --test errors out
```
% ./create_test --test
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 2817, in <module>
    sys.exit(_test())
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 2806, in _test
    m = __import__(filename[:-3])
  File "/Users/goldy/Coding/cime/scripts/Tools/../../utils/python/CIME/test_scheduler.py", line 12, in <module>
    from CIME.XML.standard_module_setup import *
ImportError: No module named CIME.XML.standard_module_setup
ERROR: Command: 'python -m doctest /Users/goldy/Coding/cime/scripts/Tools/../../utils/python/CIME/test_scheduler.py -v' failed with error 'None'

```
Seems to be machine independent

### PR DESCRIPTION
**Pull request by jedwards4b**
Fix issues with st_archive regular expression matching.  

Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit
Fixes #532 
Fixes #523 
Fixes #518 

User interface changes?: 

Code review:

### CODE REVIEW

File: cime_config/cesm/config_archive.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <comp_archive_spec compname="cam" compclass="atm">
     <rest_file_extension>\.[ri]\..*</rest_file_extension>
     <rest_file_extension>\.r[sh]\.*</rest_file_extension>
-    <hist_file_extension>\.h.*</hist_file_extension>
+    <hist_file_extension>\.h.*.nc$</hist_file_extension>
```
Comment (jedwards4b) [inline cime_config/cesm/config_archive.xml]: This prevents the files copied into suffixes for comparison from being archived

File: scripts/create_test [hunk 2]
```diff
@@ -471,8 +471,10 @@ def create_test(test_names, test_data, compiler, machine_name, no_run, no_build,
 def _main_func(description):
 ###############################################################################
     if "--test" in sys.argv:
-        CIME.utils.run_cmd_no_fail("python -m doctest %s/CIME/test_scheduler.py -v" %
```
Comment (jedwards4b) [inline scripts/create_test]: fix for 532 although no tests are actually run with this flag

File: utils/python/CIME/case_st_archive.py [hunk 3]
```diff
@@ -132,6 +132,7 @@ def _archive_history_files(case, archive, archive_entry,
 
     # determine history archive directory (create if it does not exist)
     dout_s_root = case.get_value("DOUT_S_ROOT")
+    casename = case.get_value("CASE")
```
Comment (jedwards4b) [inline utils/python/CIME/case_st_archive.py]: by explicitly including the casename we prevent suffixes matching something in that part of the string, this fixes #523

File: utils/python/CIME/user_mod_support.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ def apply_user_mods(caseroot, user_mods_path, ninst=None):
                 continue
             case_user_nl = user_nl.replace(include_dir, caseroot)
             comp = case_user_nl.split('_')[-1]
-            if ninst is not None and comp in ninst.keys():
+            if ninst is not None and comp in ninst.keys() and ninst[comp] > 1:
```
Comment (jedwards4b) [inline utils/python/CIME/user_mod_support.py]: Should not enter this code unless at least one component is multi instance.

File: cime_config/cesm/config_archive.xml (hunk 1 above)
Comment (jgfouca) [inline cime_config/cesm/config_archive.xml]: Are these changes needed on the ACME side too?

Comment (jedwards4b) [inline cime_config/cesm/config_archive.xml]: Yes, if you are using short term archiving.

**Comment by jgfouca** [comment]
Looks good, thanks. Just one question.

**Comment by jedwards4b** [comment]
Do you want me to go ahead and change the acme/config_archive.xml file?

**Comment by rljacob** [comment]
Yes, thanks.

File: cime_config/cesm/config_archive.xml (hunk 1 above)
Comment (jgfouca) [inline cime_config/cesm/config_archive.xml]: Yeah,  we do use it. Would you mind mirroring these changes on the ACME side? I know we don't like to touche each others' config files. But for bug fixes like this I think it's appropriate.

File: cime_config/acme/config_archive.xml [hunk 5]
```diff
@@ -132,33 +134,33 @@
   <comp_archive_spec compname="dwav" compclass="wav">
     <rest_file_extension>\.r.*</rest_file_extension>
     <rest_history_varname>unset</rest_history_varname>
-    <rpointer_file>rpointer$NINST_STRING.wav</rpointer_file>
```
Comment (jgfouca) [inline cime_config/acme/config_archive.xml]: It looks like these are changes that were not on the CESM side with this PR and were probably brought over when you copied the cesm file. Is this an error? Do we want these changes on ACME?

Comment (jedwards4b) [inline cime_config/acme/config_archive.xml]: I didn't see anything that would adversely affect acme - I think that these are previous updates made on the cesm side that were not made on the acme side.   When acme customizes the config_archive.xml for external acme components we will need to take care not to overwrite them, but afaict that hasn't happened yet.

**Comment by rljacob** [comment]
Does this really fix #518 ?

**Comment by jedwards4b** [comment]
No, sorry it does not - I've removed that from the list.
